autoload/go/fmt.vim

```diff
@@ -90,8 +90,12 @@ function! go#fmt#Format(withGoimport)
         let fmt_command = bin_path
     endif
 
-    " populate the final command with user based fmt options
-    let command = fmt_command . ' ' . g:go_fmt_options
+    if a:withGoimport == 1
+        let command = fmt_command
+    else
+        " populate the final command with user based fmt options
+        let command = fmt_command . ' ' . g:go_fmt_options
+    endif
```

Comment:
Can you change it to

 ```
if a:withGoimport != 1
   let command = fmt_command . ' ' . g:go_fmt_options
endif
```

And also add comment to the code why we do this, like you have wrote already in the PR description :)